cleaner/environment_matrix/classes/clean.php

```diff
@@ -96,7 +101,19 @@ static public function execute() {
                         }
 
                         if (!$dryrun) {
-                            set_config($config->config, $config->value, $config->plugin);
+
+
+                            // Build internal configuration string name such as s_auth_saml2_idpmetadata or s__passwordpolicy for core
+                            $configstring = 's_'.$config->plugin.'_'.$config->config;
```

Comment:
I'm not convinced this is 100% correct in all cases. A good test case will be a config setting which splits config into two or more form elements and then joins them back up on the server side.

Have a look at graceperiodmin which uses admin_setting_configduration

When I post this in dev tools I see two chunks of data:

s_quiz_graceperiodmin[v]: 1
s_quiz_graceperiodmin[u]: 60

We cannot make any assumptions at all around the structure of data in these raw POST data. Even if there is only a single post it could be transmitted over the network in a way which is not identical to the way it is persisted.

In fact there is no obligation for an admin setting to even use the mdl_config table at all, it could completely manage it in it's own tables. If we call the api's in the right way we should correctly support all these use cases.

For an example of this see 

https://tracker.moodle.org/browse/MDL-65818

https://github.com/sammarshallou/moodle/compare/master...MDL-65818-master#diff-a688b433b61606794abb0075b62d1a8bR2682-R2696